calc.go

```diff
@@ -27,24 +30,37 @@ func calcMoveDamage(weather Weather, user, receiver *Pokemon, move *Move) (damag
 		defense = float64(receiver.SpecialDefense())
 	}
 	// Weather modifiers
-	if weather == WeatherSandstorm {
+	if b.Weather == WeatherSandstorm {
 		if receiver.EffectiveType()&TypeRock != 0 {
 			defense *= 1.5
 		}
 		if move.Id == MoveSolarBeam {
 			movePower /= 2
 		}
 	}
-	if weather == WeatherHail && move.Id == MoveSolarBeam {
+	if b.Weather == WeatherHail && move.Id == MoveSolarBeam {
 		movePower /= 2
 	}
-	if weather == WeatherFog {
+	if b.Weather == WeatherFog {
 		if move.Id == MoveWeatherBall {
 			movePower *= 2
 		} else if move.Id == MoveSolarBeam {
 			movePower /= 2
 		}
 	}
-	damage = uint((((levelEffect * movePower * attack / defense) / 50) + 2) * modifier)
+	// Item modifiers
+	switch user.HeldItem {
+	case ItemLifeOrb:
+		modifier *= 1.30
+	case ItemMuscleBand:
+		if move.Category() == MoveCategoryPhysical {
+			modifier *= 1.10
+		}
+	case ItemWiseGlasses:
+		if move.Category() == MoveCategorySpecial {
+			modifier *= 1.10
+		}
```

Comment:
Would prefer these to be integer operations

Reply:
How do you do integer operations? Best way i've come up with is:
```golang
x *= 1.35
// in integer operation
x = (x * 135) / 100
```

Can we add some utility functions to perform floating/percentage modifiers on `uint` values?

Reply:
Yeah, that's basically it. I don't think it would be very beneficial to add utility functions just to do a multiply and divide